src/util/limits.jl

```diff
@@ -166,32 +166,30 @@ Base.:(+)(x::Limit)::Limit = x
 Base.:(-)(x::Limit)::Limit = limit(-x.val, -x.sign)
 
 #Crazy julia multiple dispatch stuff don't worry about it
-limit_types = [Int8, Int16, Int32, Int64, Int128, UInt8, UInt16, UInt32, UInt64, UInt128, BigInt, Float32, Float64]
-for S in limit_types
```

Comment:
This for loop avoids method ambiguities on types that define e.g. `max(a::MyType, b::Real)`. If we define `max(a::Real, b::Limit)` where `Limit <: Real`, then we get an ambiguity error. If we want to take your approach, we need to make `Limit` not be a subtype of `Number`. I understand that this is suboptimal.

Reply:
Oh, I see that currently `Limit <: Number`, so `Limit` is not currently subtyping `Real`. I think this might work then. But should we have `Limit <: Real`?

Reply:
This was also a random attempt to address the CI issues in continuous testing. While it makes the code cleaner, it introduces ambiguity. Since the CI now passes, I believe we no longer need this pull request. Let's close it.